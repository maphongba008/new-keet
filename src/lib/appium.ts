import { ViewProps } from 'react-native'
import _includes from 'lodash/includes'

type NamedEnum<Key extends string> = { [v in Key]: v }

function inferGeneric<T extends string>(t: NamedEnum<T>) {
  return t
}

export const APPIUM_IDs = inferGeneric({
  back_button: 'back_button',
  // LobbyScreen
  lobby_btn_join_room: 'lobby_btn_join_room',
  lobby_btn_create_room: 'lobby_btn_create_room',
  lobby_btn_create_chat: 'lobby_btn_create_chat',
  lobby_btn_create_broadcast: 'lobby_btn_create_broadcast',
  lobby_btn_create_back: 'lobby_btn_create_back',
  lobby_input_room_name: 'lobby_input_room_name',
  lobby_input_room_description: 'lobby_input_room_description',
  lobby_btn_paste: 'lobby_btn_paste',
  lobby_btn_qr: 'lobby_btn_qr',
  lobby_btn_submit: 'lobby_btn_submit',
  lobby_btn_add_room: 'lobby_btn_add_room',
  lobby_tap_to_call: 'lobby_tap_to_call',
  lobby_input_search: 'lobby_input_search',
  lobby_btn_clear_search: 'lobby_btn_clear_search',
  lobby_btn_rooms: 'lobby_btn_rooms',
  lobby_btn_profile: 'lobby_btn_profile',
  lobby_dm: 'lobby_dm',
  lobby_filter: 'lobby_filter',
  lobby_selected_filter: 'lobby_selected_filter',
  lobby_joining_rooms: 'lobby_joining_rooms',
  lobby_swipable_btn_mute: 'lobby_swipable_btn_mute',
  lobby_swipable_btn_leave: 'lobby_swipable_btn_leave',
  lobby_bottomsheet_btn_leave: 'lobby_bottomsheet_btn_leave',
  // CreateBroadcastRoomScreen
  broadcast_accept_admin: 'broadcast_accept_admin',
  broadcast_btn_create: 'broadcast_btn_create',
  // RoomScreen
  room_call_join: 'room_call_join',
  room_call_mute: 'room_call_mute',
  room_call_video: 'room_call_video',
  room_call_speaker: 'room_call_speaker',
  room_call_end: 'room_call_end',
  room_btn_attach: 'room_btn_attach',
  room_btn_send: 'room_btn_send',
  room_input_msg: 'room_input_msg',
  room_attach_doc: 'room_attach_doc',
  room_attach_media: 'room_attach_media',
  room_capture_video: 'room_capture_video',
  room_capture_img: 'room_capture_img',
  room_btn_edit: 'room_btn_edit',
  room_btn_reply: 'room_btn_reply',
  room_btn_download: 'room_btn_download',
  room_btn_copy_msg: 'room_btn_copy_msg',
  room_btn_copy_msg_link: 'room_btn_copy_msg_link',
  room_btn_copy_img: 'room_btn_copy_img',
  room_btn_share: 'room_btn_share',
  room_btn_forward: 'room_btn_forward',
  room_btn_flag: 'room_btn_flag',
  room_btn_del: 'room_btn_del',
  room_btn_clear: 'room_btn_clear',
  room_btn_invite: 'room_btn_invite',
  room_btn_pin: 'room_btn_pin',
  room_profile_name: 'room_profile_name',
  room_profile_avatar: 'room_profile_avatar',
  room_pinned_message_header: 'room_pinned_message_header',
  // RoomOptionsScreen
  options_edit_details: 'options_edit_details',
  options_btn_files: 'options_btn_files',
  options_btn_share_link: 'options_btn_share_link',
  options_btn_share_qr: 'options_btn_share_qr',
  options_btn_clear_files: 'options_btn_clear_files',
  options_btn_leave: 'options_btn_leave',
  options_btn_toggle_notifications: 'options_btn_toggle_notifications',
  options_btn_toggle_call: 'options_btn_toggle_call',
  // InviteQRScreen
  invite_btn_type: 'invite_btn_type',
  invite_btn_expires_hours: 'invite_btn_expires_hours',
  invite_btn_expires_days: 'invite_btn_expires_days',
  invite_btn_expires_weeks: 'invite_btn_expires_weeks',
  invite_btn_no_expire: 'invite_btn_no_expire',
  invite_btn_accept_admin: 'invite_btn_accept_admin',
  invite_btn_generate: 'invite_btn_generate',
  // RoomMembersScreen
  members_search: 'members_search',
  // AccountScreen
  profile_pick_avatar: 'profile_pick_avatar',
  profile_edit_name: 'profile_edit_name',
  profile_network_status: 'profile_network_status',
  account_btn_share_network_info: 'account_btn_share_network_info',
  account_btn_copy_network_info: 'account_btn_copy_network_info',
  profile_tos: 'profile_tos',
  // list
  list_item: 'list_item_' as any,
  create_channel_btn: 'create_channel_btn',
  // Avatar
  avatar_image: 'avatar_image',
  // onboarding
  onboarding_btn_skip: 'onboarding_btn_skip',
  onboarding_btn_next: 'onboarding_btn_next',
  onboarding_btn_prev: 'onboarding_btn_prev',
  onboarding_btn_finish: 'onboarding_btn_finish',
  onboarding_tos_btn_let_in: 'onboarding_tos_btn_let_in',
  onboarding_input_username: 'onboarding_input_username',
  onboarding_btn_save_username: 'onboarding_btn_save_username',
  // IdentityScreen
  identity_btn_create: 'identity_btn_create',
  identity_btn_link: 'identity_btn_link',
  identity_btn_skip: 'identity_btn_skip',
  // Room Participant list
  participant_item: 'participant_item_' as any,
  participant_btn_all: 'participant_btn_all',
  // Tooltips
  tooltip_btn_profile: 'tooltip_btn_profile',
  tooltip_btn_join_room: 'tooltip_btn_join_room',
  // PasscodeScreen
  passcode_btn_back: 'passcode_btn_back',
  passcode_btn_close: 'passcode_btn_close',
  passcode_btn_delete: 'passcode_btn_delete',
  passcode_confirm_btn_back: 'passcode_confirm_btn_back',
  passcode_confirm_btn_close: 'passcode_confirm_btn_close',
  // RecoverAccountScreen
  recover_btn_link: 'recover_btn_link',
  recover_btn_recover: 'recover_btn_recover',
  recover_input_passphrase: 'recover_input_passphrase',
  recover_btn_continue: 'recover_btn_continue',
  recover_btn_paste: 'recover_btn_paste',
  recover_btn_success: 'recover_btn_success',
  // IdentitySyncScreen
  identity_accept_checkbox: 'identity_accept_checkbox',
  identity_btn_accept: 'identity_btn_accept',
  identity_btn_use_camera_action: 'identity_btn_use_camera_action',
  identity_btn_use_invite_link: 'identity_btn_use_invite_link',
  identity_sync_invite_link_input: 'identity_sync_invite_link_input',
  identity_sync_invite_link_submit: 'identity_sync_invite_link_submit',
  identity_sync_invite_link_declined: 'identity_sync_invite_link_declined',
  identity_sync_invite_request_confirm: 'identity_sync_invite_request_confirm',
  identity_sync_invite_request_notMyDevice:
    'identity_sync_invite_request_notMyDevice',
  identity_sync_invite_request_success: 'identity_sync_invite_request_success',
  // Create ID
  create_identity_agreement_btn: 'create_identity_agreement_btn',
  create_identity_view_phrase_btn: 'create_identity_view_phrase_btn',
  create_identity_copy_phrase_btn: 'create_identity_copy_phrase_btn',
  create_identity_setup_continue_btn: 'create_identity_setup_continue_btn',
  create_id_verification_word: 'create_id_verification_word',
  create_id_submit_verification: 'create_id_submit_verification',
  create_id_onFinish_btn: 'create_id_onFinish_btn',
  // Components
  radio_btn: 'radio_btn',
  radio_btn_checked: 'radio_btn_checked',
  // Call Screen
  call_shared_screen_expand_button: 'call_shared_screen_expand_button',
  // MediaPreviewScreen
  media_preview_video_view: 'media_preview_video_view',
  media_preview_btn_custom_play: 'media_preview_btn_custom_play',
  media_preview_loading: 'media_preview_loading',
  // Discover Community
  discover_community_join_room: 'discover_community_join_room',
  discover_community_select_room: 'discover_community_select_room',

  // wallet
  wallet_default_server_action: 'wallet_default_server_action',
  wallet_custom_server_action: 'wallet_custom_server_action',
  wallet_server_get_started: 'wallet_server_get_started',
  wallet_setup_onFinish_btn: 'wallet_setup_onFinish_btn',
  wallet_create_agreement_btn: 'wallet_create_agreement_btn',
  wallet_settings_option_any: 'wallet_settings_option_any',
  wallet_settings_option_custom: 'wallet_settings_option_custom',
  wallet_settings_amount_input: 'wallet_settings_amount_input',
  wallet_settings_save_btn: 'wallet_settings_save_btn',
  wallet_send_submit_btn: 'wallet_send_submit_btn',
  wallet_send_scanner_btn: 'wallet_send_scanner_btn',
  wallet_send_scanner_back_btn: 'wallet_send_scanner_back_btn',
  wallet_send_open_scanner_btn: 'wallet_send_open_scanner_btn',
  wallet_send_paste_btn: 'wallet_send_paste_btn',
  wallet_input_wallet_address: 'wallet_input_wallet_address',
  wallet_input_transaction_description: 'wallet_input_transaction_description',
  wallet_btn_transfer_next: 'wallet_btn_transfer_next',
  wallet_btn_select_keet_contact: 'wallet_btn_select_keet_contact',
  wallet_btn_select_crypto_wallet: 'wallet_btn_select_crypto_wallet',
  wallet_receive_generate_invoice_btn: 'wallet_receive_generate_invoice_btn',
  wallet_receive_copy_invoice_btn: 'wallet_receive_copy_invoice_btn',
  wallet_receive_share_invoice_btn: 'wallet_receive_share_invoice_btn',
  wallet_custom_server_set_mode_btn: 'wallet_custom_server_set_mode_btn',
  wallet_custom_server_continue_btn: 'wallet_custom_server_continue_btn',
  wallet_custom_server_host_input: 'wallet_custom_server_host_input',
  wallet_custom_server_port_input: 'wallet_custom_server_port_input',
  wallet_custom_server_eth_indexer_input:
    'wallet_custom_server_eth_indexer_input',
  wallet_custom_server_connect_btn: 'wallet_custom_server_connect_btn',
  // User Profile Screen
  user_profile_chat: 'user_profile_chat',
  user_profile_call: 'user_profile_call',
  user_profile_send_dm: 'user_profile_send_dm',
} as const)

const DYNAMIC_KEY = [APPIUM_IDs.list_item]

export const appiumTestProps = (
  id?: keyof typeof APPIUM_IDs | string,
): Pick<ViewProps, 'accessible' | 'testID' | 'accessibilityLabel'> => {
  try {
    if (!id) {
      return {}
    }

    const appId = APPIUM_IDs[id as keyof typeof APPIUM_IDs]
    if (appId)
      return { accessible: true, testID: appId, accessibilityLabel: appId }

    if (DYNAMIC_KEY.some((substring) => _includes(id, substring))) {
      return { accessible: true, testID: id, accessibilityLabel: id }
    }
    return {}
  } catch (e) {
    console.log(e)
    return {}
  }
}
