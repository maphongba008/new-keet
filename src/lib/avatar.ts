import { ViewStyle } from 'react-native'
import b4a from 'b4a'

import { DEFAULT_AVATAR_COLORS } from './commonStyles'

export const SEED_BITS = 32
export const EMOJI_REGEXP = /\p{Emoji}/u

export const name2seed = (name: string) => {
  if (name.length === SEED_BITS) {
    return name
  } else if (name.length > SEED_BITS) {
    return name.substring(0, SEED_BITS - 1)
  }
  return name.padEnd(SEED_BITS)
}

function randInt(min: number, max: number, seed?: string): number {
  const rand = seed
    ? b4a.readUInt32LE(b4a.from(seed, 'hex')) / (2 ** SEED_BITS - 1)
    : Math.random()
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(rand * (max - min)) + min
}

export const randomColor = (seed?: string) => {
  return DEFAULT_AVATAR_COLORS[randInt(0, DEFAULT_AVATAR_COLORS.length, seed)]
}

export function getScaledFontSize({
  width,
  height,
}: Pick<ViewStyle, 'width' | 'height'>): number {
  const baseWidth = 30
  const baseHeight = 30
  const baseFontSize = 12
  const maxFontThreshold = 40

  if (typeof height !== 'number' || typeof width !== 'number') {
    return baseFontSize
  }

  // Calculate scaling factor
  const scalingFactor = (width * height) / (baseWidth * baseHeight)

  // Scale font size
  const fontSize = baseFontSize * Math.sqrt(scalingFactor)
  const minThreshold = Math.max(fontSize, baseFontSize)

  return Math.min(minThreshold, maxFontThreshold)
}
