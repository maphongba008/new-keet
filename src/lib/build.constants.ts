import Config from 'react-native-config'

const PRODUCT = {
  // define flags in alphabetical order
  FORCE_UPGRADE_BANNER: true,
  GRADIENT_TAB_BAR_ICON: true,
  IS_CHAT_CACHE_ENABLED: true,
  IS_SENTRY_ENABLED: false,
  IS_TIME_STATS_ENABLED: true,
  SHOW_ADMIN_RESET_UI: false,
  SHOW_BROADCAST_FILTER: false,
  SHOW_CHANNEL_UI: true,
  SHOW_CLEAN_DEVICE_SCREEN: false,
  SHOW_FULLSCREEN_AD: false,
  SHOW_MESSAGE_COPY_LINK: false,
  SHOW_NEW_PROFILE_FEATURE: false,
  SHOW_REMOVE_DEVICE: false,
  SHOW_RENAME_DEVICE: false,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PASSCODE: false,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: false,
  SHOW_UPDATE_BANNER: true,
  SHOW_WALLET: false,
  TRANSPARENT_TAB_BAR: true,
}

// KEEP this eslint statement or will raise lint error when swith to other setup
const STAGING = {
  ...PRODUCT,
  // define flags in alphabetical order
  IS_CHAT_CACHE_ENABLED: true,
  IS_SENTRY_ENABLED: false,
  SHOW_BROADCAST_FILTER: false,
  SHOW_CHANNEL_UI: true,
  SHOW_CLEAN_DEVICE_SCREEN: true,
  SHOW_MESSAGE_COPY_LINK: true,
  SHOW_NEW_PROFILE_FEATURE: true,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PASSCODE: true,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: true,
  SHOW_WALLET: true,
  SHOW_ADMIN_RESET_UI: true,
  SHOW_REMOVE_DEVICE: true,
  SHOW_RENAME_DEVICE: true,
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const SV = {
  ...PRODUCT,
  // define flags in alphabetical order
  GRADIENT_TAB_BAR_ICON: false,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: true,
  SHOW_UPDATE_BANNER: false,
  TRANSPARENT_TAB_BAR: false,
  IS_CHAT_CACHE_ENABLED: true,
}

// Specify a feature set alphabetically
const APP = STAGING

export const FORCE_UPGRADE_BANNER = APP?.FORCE_UPGRADE_BANNER

export const GRADIENT_TAB_BAR_ICON = APP?.GRADIENT_TAB_BAR_ICON

export const IS_SENTRY_ENABLED = APP?.IS_SENTRY_ENABLED

export const SHOW_BROADCAST_FILTER = APP?.SHOW_BROADCAST_FILTER

export const SHOW_CHANNEL_UI = APP?.SHOW_CHANNEL_UI

export const SHOW_CLEAN_DEVICE_SCREEN = APP?.SHOW_CLEAN_DEVICE_SCREEN

export const SHOW_FULLSCREEN_AD = APP?.SHOW_FULLSCREEN_AD

export const SHOW_MESSAGE_COPY_LINK = APP?.SHOW_MESSAGE_COPY_LINK

export const SHOW_NEW_PROFILE_FEATURE = APP?.SHOW_NEW_PROFILE_FEATURE

export const SHOW_PARTICIPANT_SHEET_USER_NAME =
  APP?.SHOW_PARTICIPANT_SHEET_USER_NAME

export const SHOW_PASSCODE = APP?.SHOW_PASSCODE

export const SHOW_PUSH_NOTIFICATION_OPTION = APP?.SHOW_PUSH_NOTIFICATION_OPTION

export const SHOW_ROOM_FILTER = APP?.SHOW_ROOM_FILTER

export const SHOW_UPDATE_BANNER = APP?.SHOW_UPDATE_BANNER

export const SHOW_WALLET = APP?.SHOW_WALLET

export const TRANSPARENT_TAB_BAR = APP?.TRANSPARENT_TAB_BAR

export const IS_EXPERIMENTAL_BUILD = Config?.KEET_VARIANT === 'experimental'

export const IS_CHAT_CACHE_ENABLED = APP?.IS_CHAT_CACHE_ENABLED

export const IS_TIME_STATS_ENABLED = APP?.IS_TIME_STATS_ENABLED

export const SHOW_ADMIN_RESET_UI = APP?.SHOW_ADMIN_RESET_UI

export const SHOW_REMOVE_DEVICE = APP?.SHOW_REMOVE_DEVICE

export const SHOW_RENAME_DEVICE = APP?.SHOW_RENAME_DEVICE
