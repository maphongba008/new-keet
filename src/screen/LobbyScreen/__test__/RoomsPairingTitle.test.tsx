import { fireEvent, screen } from '@testing-library/react-native'

import { APPIUM_IDs } from 'lib/appium'
import { navigate } from 'lib/navigation'
import { renderWithProviders } from 'lib/testUtils'

import RoomsPairingTitle from '../RoomsPairing.Title'

jest.mock('lib/navigation', () => ({
  ...jest.requireActual('lib/navigation'),
  navigate: jest.fn(),
}))

describe('RoomPairingTitle', () => {
  it('render correctly', async () => {
    const element = <RoomsPairingTitle />
    renderWithProviders(element, {
      preloadedState: {
        room: {
          joint: {
            pairing: {
              all: [{ id: '1' }],
            },
          },
        },
      },
    })
    expect(screen.toJSON()).toMatchSnapshot()
    fireEvent.press(screen.getByTestId(APPIUM_IDs.lobby_joining_rooms))
    expect(navigate).toHaveBeenCalledWith('pairing-rooms')
  })
  it('render correctly with no rooms pairing', async () => {
    const element = <RoomsPairingTitle />
    renderWithProviders(element, {
      preloadedState: {
        room: {
          joint: {
            pairing: {
              all: [],
            },
          },
        },
      },
    })
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
