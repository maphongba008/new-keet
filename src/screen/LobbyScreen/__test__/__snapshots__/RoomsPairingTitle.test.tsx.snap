// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomPairingTitle render correctly 1`] = `
<View
  style={
    {
      "height": {
        "callback": undefined,
        "current": 0,
        "easing": [Function],
        "onFrame": [Function],
        "onStart": [Function],
        "progress": 0,
        "reduceMotion": undefined,
        "startTime": 0,
        "startValue": 0,
        "toValue": 0,
        "type": "timing",
      },
      "opacity": {
        "callback": undefined,
        "current": 0,
        "easing": [Function],
        "onFrame": [Function],
        "onStart": [Function],
        "progress": 0,
        "reduceMotion": undefined,
        "startTime": 0,
        "startValue": 0,
        "toValue": 0,
        "type": "timing",
      },
    }
  }
>
  <View
    accessibilityLabel="lobby_joining_rooms"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#222430",
          "flexDirection": "row",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
      ]
    }
    testID="lobby_joining_rooms"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      color="#ffffff"
      fill="#ffffff"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {},
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      testID="svg-icon-clock"
      tintColor="#ffffff"
      vbHeight={24}
      vbWidth={24}
      width={16}
      xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M12 2.438A9.562 9.562 0 1 0 21.563 12 9.574 9.574 0 0 0 12 2.438m0 18A8.438 8.438 0 1 1 20.438 12 8.447 8.447 0 0 1 12 20.438M17.813 12a.563.563 0 0 1-.563.563H12a.56.56 0 0 1-.562-.563V6.75a.563.563 0 0 1 1.124 0v4.688h4.688a.563.563 0 0 1 .563.562" fill="#fff"/></svg>
"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M12 2.438A9.562 9.562 0 1 0 21.563 12 9.574 9.574 0 0 0 12 2.438m0 18A8.438 8.438 0 1 1 20.438 12 8.447 8.447 0 0 1 12 20.438M17.813 12a.563.563 0 0 1-.563.563H12a.56.56 0 0 1-.562-.563V6.75a.563.563 0 0 1 1.124 0v4.688h4.688a.563.563 0 0 1 .563.562"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "color": "#ffffff",
          "flex": 1,
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
          "includeFontPadding": false,
          "marginLeft": 12,
          "writingDirection": "ltr",
        }
      }
    >
      Joining rooms
    </Text>
    <Text
      style={
        {
          "color": "#26D2E8",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
          "includeFontPadding": false,
          "writingDirection": "ltr",
        }
      }
    >
      1
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      color="#A0A8B8"
      fill="#A0A8B8"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {},
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="svg-icon-chevronRight"
      tintColor="#A0A8B8"
      vbHeight={24}
      vbWidth={24}
      width={24}
      xml="<svg fill="currentColor" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"><path d="m17.031 12.531-7.49996 7.5c-.06968.0697-.15241.125-.24345.1627-.09105.0377-.18863.0571-.28717.0571-.09855 0-.19613-.0194-.28718-.0571-.09104-.0377-.17377-.093-.24345-.1627-.06968-.0696-.12496-.1524-.16267-.2434-.03771-.0911-.05712-.1886-.05712-.2872 0-.0985.01941-.1961.05712-.2872.03771-.091.09299-.1737.16267-.2434l6.97031-6.9694-6.97031-6.96936c-.14073-.14073-.21979-.3316-.21979-.53062 0-.19903.07906-.3899.21979-.53063s.3316-.21979.53063-.21979c.19902 0 .38989.07906.53062.21979l7.49996 7.50001c.0698.0696.1251.1524.1628.2434.0378.0911.0572.1887.0572.2872 0 .0986-.0194.1962-.0572.2872-.0377.0911-.093.1738-.1628.2434z"/></svg>
"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4288719032,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="m17.031 12.531-7.49996 7.5c-.06968.0697-.15241.125-.24345.1627-.09105.0377-.18863.0571-.28717.0571-.09855 0-.19613-.0194-.28718-.0571-.09104-.0377-.17377-.093-.24345-.1627-.06968-.0696-.12496-.1524-.16267-.2434-.03771-.0911-.05712-.1886-.05712-.2872 0-.0985.01941-.1961.05712-.2872.03771-.091.09299-.1737.16267-.2434l6.97031-6.9694-6.97031-6.96936c-.14073-.14073-.21979-.3316-.21979-.53062 0-.19903.07906-.3899.21979-.53063s.3316-.21979.53063-.21979c.19902 0 .38989.07906.53062.21979l7.49996 7.50001c.0698.0696.1251.1524.1628.2434.0378.0911.0572.1887.0572.2872 0 .0986-.0194.1962-.0572.2872-.0377.0911-.093.1738-.1628.2434z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`RoomPairingTitle render correctly with no rooms pairing 1`] = `
<View
  style={
    {
      "height": {
        "callback": undefined,
        "current": 0,
        "easing": [Function],
        "onFrame": [Function],
        "onStart": [Function],
        "progress": 0,
        "reduceMotion": undefined,
        "startTime": 0,
        "startValue": 0,
        "toValue": 0,
        "type": "timing",
      },
      "opacity": {
        "callback": undefined,
        "current": 0,
        "easing": [Function],
        "onFrame": [Function],
        "onStart": [Function],
        "progress": 0,
        "reduceMotion": undefined,
        "startTime": 0,
        "startValue": 0,
        "toValue": 0,
        "type": "timing",
      },
    }
  }
>
  <View
    accessibilityLabel="lobby_joining_rooms"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#222430",
          "flexDirection": "row",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
      ]
    }
    testID="lobby_joining_rooms"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      color="#ffffff"
      fill="#ffffff"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {},
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      testID="svg-icon-clock"
      tintColor="#ffffff"
      vbHeight={24}
      vbWidth={24}
      width={16}
      xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M12 2.438A9.562 9.562 0 1 0 21.563 12 9.574 9.574 0 0 0 12 2.438m0 18A8.438 8.438 0 1 1 20.438 12 8.447 8.447 0 0 1 12 20.438M17.813 12a.563.563 0 0 1-.563.563H12a.56.56 0 0 1-.562-.563V6.75a.563.563 0 0 1 1.124 0v4.688h4.688a.563.563 0 0 1 .563.562" fill="#fff"/></svg>
"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M12 2.438A9.562 9.562 0 1 0 21.563 12 9.574 9.574 0 0 0 12 2.438m0 18A8.438 8.438 0 1 1 20.438 12 8.447 8.447 0 0 1 12 20.438M17.813 12a.563.563 0 0 1-.563.563H12a.56.56 0 0 1-.562-.563V6.75a.563.563 0 0 1 1.124 0v4.688h4.688a.563.563 0 0 1 .563.562"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "color": "#ffffff",
          "flex": 1,
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
          "includeFontPadding": false,
          "marginLeft": 12,
          "writingDirection": "ltr",
        }
      }
    >
      Joining rooms
    </Text>
    <Text
      style={
        {
          "color": "#26D2E8",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
          "includeFontPadding": false,
          "writingDirection": "ltr",
        }
      }
    >
      0
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      color="#A0A8B8"
      fill="#A0A8B8"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {},
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="svg-icon-chevronRight"
      tintColor="#A0A8B8"
      vbHeight={24}
      vbWidth={24}
      width={24}
      xml="<svg fill="currentColor" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"><path d="m17.031 12.531-7.49996 7.5c-.06968.0697-.15241.125-.24345.1627-.09105.0377-.18863.0571-.28717.0571-.09855 0-.19613-.0194-.28718-.0571-.09104-.0377-.17377-.093-.24345-.1627-.06968-.0696-.12496-.1524-.16267-.2434-.03771-.0911-.05712-.1886-.05712-.2872 0-.0985.01941-.1961.05712-.2872.03771-.091.09299-.1737.16267-.2434l6.97031-6.9694-6.97031-6.96936c-.14073-.14073-.21979-.3316-.21979-.53062 0-.19903.07906-.3899.21979-.53063s.3316-.21979.53063-.21979c.19902 0 .38989.07906.53062.21979l7.49996 7.50001c.0698.0696.1251.1524.1628.2434.0378.0911.0572.1887.0572.2872 0 .0986-.0194.1962-.0572.2872-.0377.0911-.093.1738-.1628.2434z"/></svg>
"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4288719032,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="m17.031 12.531-7.49996 7.5c-.06968.0697-.15241.125-.24345.1627-.09105.0377-.18863.0571-.28717.0571-.09855 0-.19613-.0194-.28718-.0571-.09104-.0377-.17377-.093-.24345-.1627-.06968-.0696-.12496-.1524-.16267-.2434-.03771-.0911-.05712-.1886-.05712-.2872 0-.0985.01941-.1961.05712-.2872.03771-.091.09299-.1737.16267-.2434l6.97031-6.9694-6.97031-6.96936c-.14073-.14073-.21979-.3316-.21979-.53062 0-.19903.07906-.3899.21979-.53063s.3316-.21979.53063-.21979c.19902 0 .38989.07906.53062.21979l7.49996 7.50001c.0698.0696.1251.1524.1628.2434.0378.0911.0572.1887.0572.2872 0 .0986-.0194.1962-.0572.2872-.0377.0911-.093.1738-.1628.2434z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
